lecture-01/slices/main.go

```diff
@@ -107,3 +149,65 @@ func mergeUniqueInts(slices ...[]int) []int {
 
 	return res
 }
+
+// removeSameInts removes elements from 'dest' slice when they exist in 'source' slice
+func removeSameInts(dest, source []int) []int {
+	m := make(map[int]bool)
+
+	for _, val := range source {
+		m[val] = true
+	}
+
+	for i := 0; i < len(dest); i++ {
+		if _, ok := m[dest[i]]; ok {
+			dest, _ = removeIntAtIndex(dest, i)
+			i--
+		}
+	}
+
+	return dest
+}
+
+// shiftInts shifts all int elements of a given slice a given number of times
+// The shift happens to right in case the bool is set to true, left otherwise
+func shiftInts(s []int, i int, right bool) []int {
+	if len(s) < 2 || i >= len(s) || i <= 0 {
+		return s
+	}
+
+	if right {
+		return append(s[len(s)-i:], s[:len(s)-i]...)
+	} else {
+		return append(s[i:], s[:i]...)
+	}
+}
+
+// copyIntSlice returns a copy of the given int slice
+func copyIntSlice(src []int) []int {
+	dst := make([]int, len(src))
+	copy(dst, src)
+
+	return dst
+}
+
+// swapIntsByIndices swaps all even index elements with the nearest odd indices ones
+func swapIntsByIndices(s []int) []int {
+	for j := 1; j < len(s); j += 2 {
+		s[j-1], s[j] = s[j], s[j-1]
+	}
+
+	return s
+}
+
+// sortSlice sorts any slice of ints or strings in direct or reverse order
+func sortSlice[T int | string](s []T, reverse bool) []T {
+	sort.Slice(s, func(i, j int) bool {
```

Comment:
If you'd like to use standard libraty https://pkg.go.dev/sort#Reverse

Reply:
Yeah, know that one, was trying to play around with some handmade stuff :D